docs/react-v9/contributing/common-dev-snags.md

```diff
@@ -1,4 +1,13 @@
-// TODO #30642 A doc explaining common snags and trip hazards.
+### Cleaning your repo
 
-Assignee - Mason
-Reviewer - Sean
+Every once in a while you'll get build errors in node modules unrelated to your work.
+
+First try running:
+
+`yarn`
+
+If the issues still persist, clean your repo:
+
+`git clean -fdx`
+`yarn cache clean`
+`yarn`
```

Comment:
Although I understand the reasons for this type of docs (ie you get unexpected issues without changing packages), I think that the solution presented here is more like a symptom fix, rather than something than an actual improvement. I think that the best thing one can do in a situation like this is to either share their issue through a Teams post, or open up a repo issue.

Reply:
I disagree.

I've tried both of those things before and they both do not work to unblock me quickly enough. Both Teams messages and issues sit unanswered for at least a week.

I need to be able to unblock myself in hours, not a week.

Reply:
I understand your frustration, but these commands shouldn't be something to be followed as a rule. For example the `git clean -fdx` could mess up your progress really bad if you just blindly run the command in the terminal (ie it will remove all you untracked files which you might've missed adding). More so, if we will actually have an issue that's causing errors it's mandatory to track it so we can make sure it gets fixed for everyone.

Reply:
Sounds like this sequence is pretty commonly used among CSE-Red. This was my second time in a few months where people have recommended I do this.

Would a good compromise be to add verbiage about using caution and instructing folks on how/where to file a bug?

Reply:
I would completely avoid the `git clean -fdx` command as it's something that basically resets your whole repo (like you would delete it and clone it again). Can we just leave the `yarn` commands, please?

Reply:
Updated, removed the bit about git clean.